linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/StatusCodeStatsFilter.scala

```diff
@@ -15,7 +16,55 @@ object StatusCodeStatsFilter {
       def make(_stats: param.Stats, next: ServiceFactory[Request, Response]) = {
         val param.Stats(statsReceiver) = _stats
         if (statsReceiver.isNull) next
-        else new StatsFilter[Request](statsReceiver) andThen next
+        else new StatusCodeStatsFilter(statsReceiver).andThen(next)
       }
     }
 }
+
+/**
+ * Modified from com.twitter.finagle.http.filter.StatsFilter.
```

Comment:
I notice we don't stat response size here.  Is that intentional?

Should we just contribute memoizaiton to c.t.f.http.filter.StatsFilter upstream?